packages/preset-mini/src/_rules/typography.ts

```diff
@@ -62,6 +62,22 @@ export const fonts: Rule<Theme>[] = [
     { autocomplete: 'word-spacing-$wordSpacing' },
   ],
 
+  // stretch
+  ['font-stretch-normal', { 'font-stretch': 'normal' }],
+  ['font-stretch-ultra-condensed', { 'font-stretch': 'ultra-condensed' }],
+  ['font-stretch-extra-condensed', { 'font-stretch': 'extra-condensed' }],
+  ['font-stretch-condensed', { 'font-stretch': 'condensed' }],
+  ['font-stretch-semi-condensed', { 'font-stretch': 'semi-condensed' }],
+  ['font-stretch-semi-expanded', { 'font-stretch': 'semi-expanded' }],
+  ['font-stretch-expanded', { 'font-stretch': 'expanded' }],
+  ['font-stretch-extra-expanded', { 'font-stretch': 'extra-expanded' }],
+  ['font-stretch-ultra-expanded', { 'font-stretch': 'ultra-expanded' }],
+  [
+    /^(?:font-)?stretch-(.+)$/,
```

Comment:
```suggestion
    /^font-stretch-(.+)$/,
```

I guess let's do it strictly first and later we could discuss if it's worth to provide the shorthand. Also align better with the static ones